site/lib/src/components/pages/changelog/changelog_filters.dart

```diff
@@ -0,0 +1,157 @@
+
+    final entriesToShow = filters.filterEntries(entries, searchQuery);
+    filteredEntriesCount = entriesToShow.length;
+
+    final listContainer = web.document.getElementById('all-changelog-list');
+    if (listContainer != null) {
+      final entryCards = listContainer.querySelectorAll('.changelog-card');
+      for (var i = 0; i < entryCards.length; i++) {
+        final element = entryCards.item(i) as web.HTMLElement;
+        // Use index to find the corresponding original entry object.
+        if (i < entries.length) {
+          final entry = entries[i];
+          if (entriesToShow.contains(entry)) {
+            element.classList.remove('hidden');
+          } else {
+            element.classList.add('hidden');
+          }
+        }
+      }
+    }
+  }
+
+  @override
+  void dispose() {
+    if (kIsWeb) {
+      filters.removeListener(setFilters);
+    }
+    super.dispose();
+  }
+
+  @override
+  Component build(BuildContext context) {
+    return FilterToolbar(
+      id: 'changelog-search-group',
+      searchBar: SearchBar(
+        placeholder: 'Search changelog...',
+        label: 'Search changelog',
+        value: searchQuery,
+        onInput: (value) {
+          setFilters(() {
+            searchQuery = value;
+          });
+        },
+      ),
+      resultCount: label(
+        attributes: {'for': 'changelog-search'},
+        [
+          const .text('Showing '),
+          span([.text('$filteredEntriesCount')]),
+          const .text(' / '),
+          span([.text('${entries.length}')]),
+        ],
+      ),
+      actions: [
+        button(
+          classes: 'icon-button show-filters-button',
+          onClick: () {
+            final toggle =
+                web.document.getElementById('open-filter-toggle')
+                    as web.HTMLInputElement?;
+            if (toggle != null) {
+              toggle.checked = !toggle.checked;
+            }
+          },
+          [
+            const span(classes: 'material-symbols-outlined', [
+              .text('filter_list'),
+            ]),
+          ],
+        ),
+      ],
+      bottomActions: [
+        button(
+          attributes: {
+            if (searchQuery.isEmpty &&
+                filters.selectedTags.isEmpty &&
+                filters.selectedAreas.isEmpty &&
+                filters.selectedVersions.isEmpty)
+              'disabled': 'true',
+          },
+          onClick: () {
+            // Update search query and reset filters.
+            // We call setFilters to ensure the UI updates even if
+            // filters.reset() doesn't trigger a change,
+            // such as if only search query was active.
+            searchQuery = '';
+            filters.reset();
+            setFilters();
+          },
+          [
+            const span(classes: 'material-symbols-outlined', [
+              .text('close_small'),
+            ]),
+            const span([.text('Clear filters')]),
+          ],
+        ),
```

Comment:
Can reuse our existing `Button` component here to simplify the implementation and avoid reimplementing anything:


```suggestion
        Button(
          icon: 'close_small',
          content: 'Clear filters',
          disabled:
              searchQuery.isEmpty &&
              filters.selectedTags.isEmpty &&
              filters.selectedAreas.isEmpty &&
              filters.selectedVersions.isEmpty,
          onClick: () {
            // Update search query and reset filters.
            // We call setFilters to ensure the UI updates even if
            // filters.reset() doesn't trigger a change,
            // such as if only search query was active.
            searchQuery = '';
            filters.reset();
            setFilters();
          },
        ),
```